File: mercado_backend/src/main/java/com/mercado/pagos/servicios/service/PuestoService.java
package com.mercado.pagos.servicios.service;

import com.mercado.pagos.servicios.dto.response.PageResponseDTO;
import com.mercado.pagos.servicios.dto.response.PuestoResponseDTO;

public interface PuestoService {

    PuestoResponseDTO crearPuesto();

    PageResponseDTO<PuestoResponseDTO> listarPuestos(boolean sinAsignacion, int page, int size);

}

File: mercado_frontend/src/app/features/puestos/services/puesto.service.ts
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from '../../../core/services/api.service';
import { PageResponseDTO, PuestoRequestDTO, PuestoResponseDTO } from '../models/puesto.models';

@Injectable({
  providedIn: 'root'
})
export class PuestoService {
  private readonly api = inject(ApiService);
  private readonly basePath = '/puestos';

  listar(page: number, size: number, sinAsignacion = false): Observable<PageResponseDTO<PuestoResponseDTO>> {
    return this.api.get<PageResponseDTO<PuestoResponseDTO>>(this.basePath, { page, size, sinAsignacion });
  }

  crear(): Observable<PuestoResponseDTO> {
    const request: PuestoRequestDTO = {};
    return this.api.post<PuestoResponseDTO, PuestoRequestDTO>(this.basePath, request);
  }
}
